### PR TITLE
Migrate from node_modules to yarn PnP

### PR DESCRIPTION
**Pull request by xdaniortega**

### CODE REVIEW

**Comment by xdaniortega** [comment]
When calling `yarn proofs:setup` script, it runs ` "fmt:write` and fails throwing following error:
```
[error] Cannot find package 'prettier-plugin-solidity' imported from /Users/danielortega/Desktop/cartesi/yarn-update/rollups-contracts/onchain/rollups/noop.js
```
**TROUBLES FOUND**
There are two possible solutions here:
1. Incorrect dependencies in packages
Install prettier with command: `npm i prettier-plugin-solidity`
But fails with error:
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: @cartesi/rollups@1.0.0
npm ERR! Found: ethers@5.7.2
npm ERR! node_modules/ethers
npm ERR!   dev ethers@"^5" from @cartesi/rollups@1.0.0
npm ERR!   rollups
npm ERR!     @cartesi/rollups@1.0.0
npm ERR!     node_modules/@cartesi/rollups
npm ERR!       workspace rollups from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer ethers@"6.x" from @typechain/ethers-v6@0.4.3
npm ERR! node_modules/@typechain/ethers-v6
npm ERR!   peer @typechain/ethers-v6@"^0.4.3" from @typechain/hardhat@8.0.3
npm ERR!   node_modules/@typechain/hardhat
npm ERR!     dev @typechain/hardhat@"^8" from @cartesi/rollups@1.0.0
npm ERR!     rollups
npm ERR!       @cartesi/rollups@1.0.0
npm ERR!       node_modules/@cartesi/rollups
npm ERR! 
```
2. Incorrect editor configuration.
At yarn pnp [guide](https://yarnpkg.com/migration/pnp) we can find:
`Make sure that typescript, eslint, prettier, ... all dependencies typically used by your IDE extensions are listed at the top level of the project (rather than in an arbitrary workspace).`

But fails with the exact same input as before after running:

- Install the [ZipFS](https://marketplace.visualstudio.com/items?itemName=arcanis.vscode-zipfs) VSCode extension.
- Run yarn [dlx](https://yarnpkg.com/cli/dlx) @yarnpkg/sdks vscode and commit the changes.
- For TypeScript, don't forget to select [Use Workspace Version](https://code.visualstudio.com/docs/typescript/typescript-compiling#_using-the-workspace-version-of-typescript) in VSCode.

**Comment by guidanoli** [comment]
> Install the [ZipFS](https://marketplace.visualstudio.com/items?itemName=arcanis.vscode-zipfs) VSCode extension.
> Run yarn [dlx](https://yarnpkg.com/cli/dlx) @yarnpkg/sdks vscode and commit the changes.
> For TypeScript, don't forget to select [Use Workspace Version](https://code.visualstudio.com/docs/typescript/typescript-compiling#_using-the-workspace-version-of-typescript) in VSCode.

We should avoid relying on any specific IDE.

File: .gitmodules
Comment (guidanoli) [inline .gitmodules:1]: The changes in this file can be reverted.

File: onchain/rollups/package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
         "info": "npm-scripts-info",
         "proofs:update": "./test/foundry/dapp/helper/update-proofs.sh",
         "proofs:setup": "./test/foundry/dapp/helper/update-proofs.sh --setup",
-        "prepack": "run-s build tsc:prod copy-dts",
+        "prestart": "run-s build tsc:prod copy-dts",
```
Comment (guidanoli) [inline onchain/rollups/package.json:27]: I don't understand this change.

**Comment by guidanoli** [comment]
Shouldn't this PR be based on top of #75?

File: onchain/rollups/package.json (hunk 1 above)
Comment (xdaniortega) [reply onchain/rollups/package.json:27]: You're right, `prestart` is not neither defining better nor is a better use in yarn pnp. In fact, all `Pre` and `post` scripts have been deliberately deprecated, we should avoid them in a near future. We can tackle this in the next iteration.

**Comment by xdaniortega** [comment]
> Shouldn't this PR be based on top of #75?

Not really, because #75 is ready to be merged as the upgrade from 1.x to 3.x is ready and fully compatible and this PR have several more implications and we decided to not implement it for the next release. It will have to be rebase on top of main after we decide to fully upgrade yarn PnP.

**Comment by guidanoli** [comment]
> Not really, because #75 is ready to be merged as the upgrade from 1.x to 3.x is ready and fully compatible and this PR have several more implications and we decided to not implement it for the next release. It will have to be rebase on top of main after we decide to fully upgrade yarn PnP.

You can safely change the base branch of this PR to that of #75, because this allows this PR to show only the new commits that sit on top of #75, which helps us review your code. :-)
This is the magic of [stacked PRs](https://blog.logrocket.com/using-stacked-pull-requests-in-github/).

**Pull request closed without merging**